File: src/game/tools/rpc/output/wrap/AchievementListWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class AchievementListWrap : IWrapper
	{	
		public AchievementWrap[] AchList; // 成就列表
		
		public void Decode(Packet pck)
		{	
			AchList = new AchievementWrap[pck.GetShort()];
			for (int i = 0; i < AchList.Length; i++)
			{
				AchList[i] = new AchievementWrap();
				AchList[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	if (AchList == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)AchList.Length);
	        	for(int i = 0; i < AchList.Length; i++)
	        	{
	        		AchList[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}

File: src/game/tools/rpc/output/wrap/AchievementListWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type AchievementListWrap struct {
	
	AchList []*AchievementWrap // 成就列表
	rpc.Wrapper
}

func (w *AchievementListWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.AchList = make([]*AchievementWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.AchList); i++ {
		w.AchList[i] = new(AchievementWrap)
		w.AchList[i].Decode(pck)
		
	}
	
	return w
}

func (w *AchievementListWrap)Encode(pck *rpc.Packet) {
	
	if w.AchList == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.AchList)))
    	for i := 0; i < len(w.AchList); i++ {
    		w.AchList[i].Encode(pck);
			
    	}
    }
	
}

